src/sync/installation.test.ts

```diff
@@ -357,4 +360,55 @@ describe("sync/installation", () => {
 		});
 	});
 
+	describe("Update job status", () => {
+		const GITHUB_INSTALLATION_ID = 1111;
+		const REPO_ID = 12345;
+		const commitsFromDate = new Date();
+		let data: BackfillMessagePayload;
+		let sub: Subscription;
+		let repoSync: RepoSyncState;
+		beforeEach(async ()=> {
+			sub = await Subscription.install({ host: jiraHost, installationId: GITHUB_INSTALLATION_ID, gitHubAppId: undefined, hashedClientKey: "client-key" });
+			repoSync = await RepoSyncState.create({
+				subscriptionId: sub.id, repoId: REPO_ID, repoName: "name", repoUrl: "url", repoOwner: "owner", repoFullName: "full name",
+				repoPushedAt: new Date(), repoUpdatedAt: new Date(), repoCreatedAt: new Date()
+			});
+			data = {
+				installationId: GITHUB_INSTALLATION_ID,
+				jiraHost: jiraHost,
+				commitsFromDate: commitsFromDate.toISOString()
+			};
+		});
+		it("should update backfill from date upon success complete and existing backfill date is empty", async () => {
+			await updateJobStatus(data, { edges: [], jiraPayload: undefined }, "pull", REPO_ID, getLogger("test"), jest.fn());
+			await repoSync.reload();
+			expect(repoSync.pullFrom!.toISOString()).toEqual(commitsFromDate.toISOString());
+		});
+		it("should update backfill from date upon success complete and new backfill date is earlier", async () => {
+			repoSync.pullFrom = new Date(commitsFromDate.getTime() + 100000);
+			await repoSync.save();
+			await updateJobStatus(data, { edges: [], jiraPayload: undefined }, "pull", REPO_ID, getLogger("test"), jest.fn());
+			await repoSync.reload();
+			expect(repoSync.pullFrom!.toISOString()).toEqual(commitsFromDate.toISOString());
+		});
+		it("should skip update backfill from date upon success complete and new backfill date is more recent", async () => {
+			const oldDate = new Date(commitsFromDate.getTime() - 100000);
+			repoSync.pullFrom = oldDate;
+			await repoSync.save();
+			await updateJobStatus(data, { edges: [], jiraPayload: undefined }, "pull", REPO_ID, getLogger("test"), jest.fn());
+			await repoSync.reload();
+			expect(repoSync.pullFrom!.toISOString()).toEqual(oldDate.toISOString());
+		});
+		it("should skip update backfill from date if task is branch", async () => {
+			await updateJobStatus(data, { edges: [], jiraPayload: undefined }, "branch", REPO_ID, getLogger("test"), jest.fn());
+			await repoSync.reload();
+			expect(repoSync.branchFrom).toBeNull();
+		});
+		it("should nott update backfill from date is job is not complete", async () => {
+			await updateJobStatus(data, { edges: [ { cursor: "abcd" } ], jiraPayload: undefined }, "pull", REPO_ID, getLogger("test"), jest.fn());
+			await repoSync.reload();
+			expect(repoSync.pullFrom).toBeNull();
+		});
+	});
+
```

Comment:
Should we add more test cases to include other tasks not just `pull` and `branch`?

Reply:
fixed.